sirepo/package_data/static/js/srw.js

```diff
@@ -807,6 +807,82 @@ SIREPO.app.controller('BeamlineController', function (activeSection, appState, b
     });
 });
 
+SIREPO.app.controller('MLController', function (appState, panelState, persistentSimulation, requestSender, srwService, $scope, $window) {
+    const self = this;
+    self.appState = appState;
+    self.srwService = srwService;
+    self.simScope = $scope;
+    self.resultsFile = null;
+    self.simComputeModel = 'machineLearningAnimation';
+    self.simState = persistentSimulation.initSimulationState(self);
+
+    function openActivait(data) {
+        requestSender.sendStatelessCompute(
+            appState,
+            d => {
+                $window.location.href = `activait#/data/${data.models.simulation.simulationId}`;
+            },
+            {
+                method: 'results_to_activait',
+                args: {
+                    file: self.resultsFile,
+                    id: appState.models.simulation.simulationId,
+                }
+            }
+        );
+    }
+
+    self.createActivaitSimulation = () => {
```

Comment:
I see the problem. createActivaitSimulation calls newSimulation. There's no need for openActivait to be called here at all. Rather, implement this in template.srw.new_simulation. See the code for api_newSimulation. This is documented in https://github.com/radiasoft/sirepo/wiki/Template-API. Might be worth a read.

This comes down to the rule: If you find yourself writing business logic in the GUI, something is wrong. The GUI is about presentation, and this is business logic about how simulations are created.

Reply:
[93f26d2](https://github.com/radiasoft/sirepo/pull/5216/commits/93f26d2abc840faddd2fc8d2684fad38c0908736)